mvc/views/view-components.rst

```diff
@@ -21,54 +30,98 @@ New to ASP.NET MVC 6, view components are similar to partial views, but they are
+See `ViewComponent properties <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/ViewComponent/index.html#properties>`__ and `ViewComponent Class <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/ViewComponent/index.html>`__.
 
-A view component consists of two parts, the class (typically derived from  ``ViewComponent``) and the Razor view which calls methods in the view component class. Like controllers, a view component can be a POCO, but most users will want to take advantage of the methods and properties available by deriving from ``ViewComponent``.
+Creating a view component
+---------------------------
+
+This section contains the high level requirements to create a view component. Later in the article we'll examine each step in detail and create a view component.
+
+The view component class
+^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 A view component class can be created by any of the following:
 
-- Deriving from `ViewComponent`.
-- Decorating the class with the ``[ViewComponent]`` attribute, or deriving from a class with the ``[ViewComponent]`` attribute.
-- Creating a class where the name ends with the suffix *ViewComponent*.
+- Deriving from `ViewComponent`
+- Decorating a class with the ``[ViewComponent]`` attribute, or deriving from a class with the ``[ViewComponent]`` attribute
+- Creating a class where the name ends with the suffix *ViewComponent*
 
-Like controllers, view components must be public, non-nested, non-abstract classes.
+Like controllers, view components must be public, non-nested, and non-abstract classes.
 
-Examine the ViewComponent class
---------------------------------
+View component methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-- Examine the *src\\TodoList\\ViewComponents\\PriorityListViewComponent.cs* file:
+- Contain ``Invoke`` or ``InvokeAsync`` methods that return an ``IViewComponentResult``
+- Fully support constructor `Dependency Injection <https://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`__
+- Typically initialize a model and pass it to a view by calling the ``ViewComponent`` ``View`` method. 
+- Parameters come from the calling method, not HTTP, there is no model binding. This allows you to overload ``Invoke``/``InvokeAsync``  (which you can't do with controllers).
+- Do not take part in the controller lifecycle, which means you can't use action filters in a view component
+- Are not reachable directly as an HTTP endpoint, they are invoked from your code (usually in a view). A View Component never handles a request.
+- Are overloaded on the signature rather than any details from the current HTTP request
 
-  .. code-block:: c#
-    :linenos:
+View search path
```

Comment:
I think you need a little bit of prose to describe what this section is about. Something like: 

**View search path**

The runtime searches for views for view components in the following paths:
- `Views/<controller_name>/Components/<view_component_name>`
- `Views/Shared/Components/<view_component_name>`

The default view name for a view component is *Default*, which means your view file will typically be named *Default.html*. You can specify a different view name when creating the view component result or when calling the `View` method.